### PR TITLE
fix: Truncate text in spinwidget

### PR DESCRIPTION
**Pull request by robert00s**
Before:

After:

### CODE REVIEW

File: frontend/ui/widget/spinwidget.lua [hunk 1]
```diff
@@ -83,12 +84,23 @@ function SpinWidget:update()
         value_widget,
     }
 
+    local close_button = CloseButton:new{ window = self, padding_top = Size.margin.title, }
+    local btn_width = close_button:getSize().w + Size.padding.default * 2
+    local title_txt_width = RenderText:sizeUtf8Text(
+        0, self.width, self.title_face, self.title_text).x
+    local show_title_txt
+    if self.width < (title_txt_width + btn_width) then
+        show_title_txt = RenderText:truncateTextByWidth(
```
Comment (Frenzie) [inline frontend/ui/widget/spinwidget.lua]: Is there any hold to show full text implementation in place?

Comment (robert00s) [reply frontend/ui/widget/spinwidget.lua]: What do you mean exactly? :)

Comment (Frenzie) [reply frontend/ui/widget/spinwidget.lua]: Like how on most things you can hold or something to show the full text.

Comment (robert00s) [reply frontend/ui/widget/spinwidget.lua]: But we didn't do that on title bar earlier. I'm not sure is very useful.

Comment (Frenzie) [reply frontend/ui/widget/spinwidget.lua]: Just wondering. :-)

**Pull request merged**